pkg/scaffold/v2/main.go

```diff
@@ -79,7 +79,15 @@ func main() {
 
 	ctrl.SetLogger(zap.Logger(true))
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{ Scheme: scheme, MetricsBindAddress: metricsAddr })
+	// Persmissions to do leader election.
+	// +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+	// +kubebuilder:rbac:groups="",resources=configmaps/status,verbs=get;update;patch
```

Comment:
> I remember some conversation about using a resource for leader election instead of configmaps

It is available on newer cluster (I will need to check from what version). We are still using configmap in our CR library. Ref: https://github.com/kubernetes-sigs/controller-runtime/blob/6fc7baa65f8966f45353903c0e886fe23a21af73/pkg/leaderelection/leader_election.go#L82

Reply:
We can probably switch to resource based leader-election in the next release of controller-runtime.